### PR TITLE
Cl 8 ys read only feature

### PR DESCRIPTION
**Pull request by Yash-Shukla-9**
Firstly merged the cl-6-Auth branch to this current branch cl-8 ReadOnly to get the Auth setup from that branch.
created a flag for ReadOnly feature.

### CODE REVIEW

File: client/src/main.jsx [hunk 1]
```diff
@@ -24,4 +25,14 @@ const router = createBrowserRouter([
 const rootElement = document.getElementById("root");
 const root = createRoot(rootElement);
 
-root.render(<RouterProvider router={router} />);
+root.render(
+  <Auth0Provider
+    domain="dev-gh7vd2ytg45l280b.us.auth0.com"
```
Comment (xDAnkit) [inline client/src/main.jsx]: Move this inside env file

File: client/src/NavBar.jsx [hunk 2]
```diff
@@ -136,15 +149,30 @@ export default function NavBar(props) {
...
+          ) : (
+            <div className="Profile">
+              <div className="Profile-Box">
+                {isAuthenticated && <p className="ProfileName">{user.name}</p>}
+              </div>
+
+              <button
+                type="button"
+                className="m-1 btn btn-danger btn-sm"
+                onClick={() => logout()}
```
Comment (xDAnkit) [inline client/src/NavBar.jsx]: use `onClick={logout}`

**Review by xDAnkit — COMMENTED** [review]
Check my comments

File: client/src/NavBar.jsx (hunk 2 above)
Comment (Yash-Shukla-9) [reply client/src/NavBar.jsx]: refactor: remove the arrow function

File: client/src/main.jsx (hunk 1 above)
Comment (Yash-Shukla-9) [reply client/src/main.jsx]: build: new env file and moved all the variables into it.

**Comment by Yash-Shukla-9** [comment]
> Check my comments

**Comment by Yash-Shukla-9** [comment]
Fix: all issue is resolved please review.

File: client/.env [hunk 3]
```diff
@@ -0,0 +1,2 @@
+VITE_AUTH_DOMAIN=dev-gh7vd2ytg45l280b.us.auth0.com
```
Comment (xDAnkit) [inline client/.env]: @Yash-Shukla-9 - We should never commit our secure keys

File: client/src/EditorTab.jsx [hunk 4]
```diff
@@ -2,13 +2,18 @@ import Editor from "@monaco-editor/react";
...
+import { useAuth0 } from "@auth0/auth0-react";
 export default function EditorTab() {
   const Lang = useOutletContext();
   const location = useLocation();
   const [codeSave, setCodeSave] = useState(null);
   const [editorCode, setEditorCode] = useState("");
-  console.log(location.pathname.split("/")[2]);
-  console.log(Lang);
+
+  // getting th flag
```
Comment (xDAnkit) [inline client/src/EditorTab.jsx]: @Yash-Shukla-9 - Please fix your spellings

File: client/src/EditorTab.jsx [hunk 5]
```diff
@@ -2,13 +2,18 @@ import Editor from "@monaco-editor/react";
...
+  const { user, isAuthenticated } = useAuth0();
+  const hasUser = isAuthenticated && user.name === "Yash Shukla";
+
+  // console.log(location.pathname.split("/")[2]);
```
Comment (xDAnkit) [inline client/src/EditorTab.jsx]: @Yash-Shukla-9 - Remove unwanted logs

File: client/src/main.jsx [hunk 6]
```diff
@@ -2,9 +2,18 @@ import { StrictMode } from "react";
...
 import "bootstrap/dist/css/bootstrap.css";
+import { Auth0Provider } from "@auth0/auth0-react";
 import App from "./App";
 import EditorTab from "./EditorTab";
 import HomePage from "./HomePage";
+
+const domain = import.meta.env.VITE_AUTH_DOMAIN;
+const clientId = import.meta.env.VITE_AUTH_CLIENT_ID;
+
+// console.log(domain, clientId);
```
Comment (xDAnkit) [inline client/src/main.jsx]: @Yash-Shukla-9 - Remove unwanted logs

File: client/src/main.jsx [hunk 7]
```diff
@@ -2,9 +2,18 @@ import { StrictMode } from "react";
 import { createRoot } from "react-dom/client";
 import { createBrowserRouter, RouterProvider } from "react-router-dom";
 import "bootstrap/dist/css/bootstrap.css";
+import { Auth0Provider } from "@auth0/auth0-react";
 import App from "./App";
 import EditorTab from "./EditorTab";
 import HomePage from "./HomePage";
+
+const domain = import.meta.env.VITE_AUTH_DOMAIN;
+const clientId = import.meta.env.VITE_AUTH_CLIENT_ID;
```
Comment (xDAnkit) [inline client/src/main.jsx]: @Yash-Shukla-9 Can we move these variables to some other file to maintain data abstractions

Comment (xDAnkit) [inline client/src/main.jsx]: @Yash-Shukla-9 Can we move these variables to some other file to maintain data abstractions

Comment (xDAnkit) [inline client/src/main.jsx]: @Yash-Shukla-9 Can we move these variables to some other file to maintain data abstractions.

**Review by xDAnkit — CHANGES_REQUESTED** [review]
@Yash-Shukla-9 - Please check my comments

File: client/.env [hunk 8]
```diff
@@ -0,0 +1,2 @@
...
+VITE_AUTH_CLIENT_ID=s9hSftlUTUPedPcLx0xG5Urh186qqt3U
```
Comment (Yash-Shukla-9) [inline client/.env]: okay sir but in this all the keys are placed in the .env file

File: client/src/main.jsx (hunk 7 above)
Comment (Yash-Shukla-9) [reply client/src/main.jsx]: Fix: created a  file for these kind off variables

**Review by Yash-Shukla-9 — COMMENTED** [review]
Fix: all the issue which were mentioned

File: client/package.json [hunk 9]
```diff
@@ -27,6 +28,8 @@
     "eslint-plugin-react": "^7.33.2",
     "eslint-plugin-react-hooks": "^4.6.0",
     "eslint-plugin-react-refresh": "^0.4.4",
+    "install": "^0.13.0",
+    "npm": "^10.2.5",
```
Comment (xDAnkit) [inline client/package.json]: @Yash-Shukla-9 - Remove this

File: client/src/NavBar.jsx [hunk 10]
```diff
@@ -82,29 +83,41 @@ const supportedLangList = {
...
+        .then((res) => {
+          const selectedKey = Object.keys(supportedLangList).find(
+            (key) => res.data.language == supportedLangList[key]
+          );
+          setSelectLang(selectedKey);
+        });
     }
-  },[]);
+  }, []);
   function onSelectLang(value) {
```
Comment (xDAnkit) [inline client/src/NavBar.jsx]: @Yash-Shukla-9 - Change this ti arrow func

**Review by xDAnkit — APPROVED** [review]
Looks good to me for now : But need improvements

**Pull request merged**